nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/operation/GetContainerRendition.java

```diff
@@ -0,0 +1,173 @@
+                        break;
+                    }
+                }
+            }
+        }
+        return blobs;
+    }
+
+    protected Blob getDefaultRendition(DocumentModel doc) {
+        Blob blob = null;
+        if ((collectionManager.isCollection(doc) || doc.hasFacet(FacetNames.FOLDERISH)) && currentDepth < maxDepth) {
+            blob = processContainer(doc);
+        } else {
+            Rendition rendition = renditionService.getDefaultRendition(doc, reason, null);
+            if (rendition != null) {
+                blob = rendition.getBlob();
+                if (blob == null) {
+                    log.debug(String.format("Default rendition '%s' has an null Blob for document '%s'",
+                            rendition.getName(), doc.getPathAsString()));
+                }
+            }
+        }
+        return blob;
+    }
+
+    private BlobList getFolderishBlobs(DocumentModel parent) {
+        BlobList blobs = new BlobList();
+        DocumentModelIterator it = session.getChildrenIterator(parent.getRef());
+        while (it.hasNext()) {
+            DocumentModel child = it.next();
+            if (!LifeCycleConstants.DELETED_STATE.equals(session.getCurrentLifeCycleState(child.getRef()))) {
+                Blob blob = getDefaultRendition(child);
+                if (blob != null) {
+                    blobs.add(blob);
+                    if (limit > -1 && ++added >= limit) {
+                        break;
+                    }
+                }
+            }
+        }
+        return blobs;
+    }
+
+    protected Blob processContainer(DocumentModel doc) {
+        AutomationService as = Framework.getLocalService(AutomationService.class);
+        try (OperationContext oc = new OperationContext(session)) {
+            oc.push("limit", limit);
+            oc.push("maxDepth", maxDepth);
+            oc.push("_currentDepth", currentDepth + 1);
+            oc.push("reason", reason);
+            oc.setInput(doc);
+            return (Blob) as.run(oc, ID);
+        } catch (OperationException e) {
+            throw new NuxeoException(String.format("Could not compute rendition for %s", doc.getPathAsString()));
+        }
+    }
+
+    @OperationMethod
+    public Blob run(DocumentModel doc) {
+        BlobList blobs = null;
+        if (collectionManager.isCollection(doc)) {
+            blobs = getCollectionBlobs(doc.getAdapter(Collection.class));
+        } else if (doc.hasFacet("Folderish")) {
+            blobs = getFolderishBlobs(doc);
+        } else {
+            log.warn("The operation only accepts folderish document or collection");
+            return null;
+        }
+        AutomationService as = Framework.getLocalService(AutomationService.class);
+        try (OperationContext oc = new OperationContext(session)) {
+            oc.push("filename", doc.getName());
+            oc.setInput(blobs);
+            return (Blob) as.run(oc, CreateZip.ID);
```

Comment:
You should probably call the zip code directly instead of going through the Automation layer.